### PR TITLE
Updating group name to scheduling.x-k8s.io

### LINKED ISSUE #973: Update group name to scheduling.x-k8s.io
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind feature

**What happened**:

It's better to follow the practice for group name of k8s-sigs , e.g. `x-k8s.io`.

### PR DESCRIPTION
**Pull request by sriramg96**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #973 

**Special notes for your reviewer**: N/A

**Release note**:

```None

```

### CODE REVIEW

**Comment by sriramg96** [comment]
> [CLA Not Signed](https://api.easycla.lfx.linuxfoundation.org/v2/repository-provider/github/sign/20677499/95838111/974/#/?version=2)
> 
> * [❌](https://api.easycla.lfx.linuxfoundation.org/v2/repository-provider/github/sign/20677499/95838111/974/#/?version=2) - login: @sriramg96 / name: Sriram Ganesan . The commit ([e806d6c](https://github.com/kubernetes-sigs/kube-batch/commit/e806d6cfde1e8cd4a7372590aa46e9a383ddc836)) is not authorized under a signed CLA. [Please click here to be authorized](https://api.easycla.lfx.linuxfoundation.org/v2/repository-provider/github/sign/20677499/95838111/974/#/?version=2). For further assistance with EasyCLA, [please submit a support request ticket](https://jira.linuxfoundation.org/servicedesk/customer/portal/4).

/easycla

File: pkg/apis/scheduling/v1alpha1/labels.go [hunk 1]
```diff
@@ -18,11 +18,11 @@ package v1alpha1
 
 // GroupNameAnnotationKey is the annotation key of Pod to identify
 // which PodGroup it belongs to.
-const GroupNameAnnotationKey = "scheduling.k8s.io/group-name"
+const GroupNameAnnotationKey = "scheduling.x-k8s.io/group-name"
```
Comment (sriramg96) [inline pkg/apis/scheduling/v1alpha1/labels.go:21]: @k82cn as part of [this issue](https://github.com/kubernetes-sigs/kube-batch/issues/973) did we want to also update the annotation keys as such or was it just the group names? Just wanted to check/clarify

Comment (k82cn) [reply pkg/apis/scheduling/v1alpha1/labels.go:21]: let's update the annotation key accordingly to keep them align :)

**Comment by sriramg96** [comment]
@k82cn I think the generated doc files need to be updated (such as [here](https://github.com/kubernetes-sigs/kube-batch/blob/master/vendor/k8s.io/api/scheduling/v1alpha1/types_swagger_doc_generated.go#L31)), but I don't see this `hack/update-generated-swagger-docs.sh` file that was used to generated the swagger docs. I was wondering if you knew if this was renamed or moved elsewhere, and if so how to regenerate the docs.

**Comment by k82cn** [comment]
> @k82cn I think the generated doc files need to be updated (such as [here](https://github.com/kubernetes-sigs/kube-batch/blob/master/vendor/k8s.io/api/scheduling/v1alpha1/types_swagger_doc_generated.go#L31)), but I don't see this `hack/update-generated-swagger-docs.sh` file that was used to generated the swagger docs. I was wondering if you knew if this was renamed or moved elsewhere, and if so how to regenerate the docs.

I'm not sure about that part. Maybe we can handle it later :)

**Pull request closed without merging**